itest/utils.go

```diff
@@ -61,6 +62,51 @@ type ClientEventStream[T any] interface {
 type EventSubscription[T any] struct {
 	ClientEventStream[T]
 	Cancel context.CancelFunc
+
+	// Filter is an optional filter function that can be used to filter
+	// events received from the client stream.
+	//
+	// If set, it will be called for each event received from the stream. If
+	// it returns true, the event is returned. If it returns false, the
+	// event is ignored and the next event is received from the stream.
+	Filter func(T) (bool, error)
```

Comment:
Same here re naming. Should we call this `ShouldNotify`? Or whatever we choose to call the function(s) in `rpcserver`.